packages/modules/auth/lib/store/index.js

```diff
@@ -151,6 +151,59 @@ const ACTIONS = {
       ...(signInRedirectRouteName ? { signInRedirectRouteName } : {}),
       ...(signUpRouteName ? { signUpRouteName } : {})
     });
+  },
+
+  changePassword({ dispatch }, payload) {
+    const { initiator, data: formPass } = payload;
+    let oldPrivateKey;
+
+    try {
+      if (deipRpc.auth.isWif(formPass.oldPassword)
+      && deipRpc.auth.wifToPublic(initiator.privKey) === initiator.signUpPubKey) {
+        // if old private key is entered
+
+        oldPrivateKey = initiator.privKey;
+      } else {
+        // if old password is entered or old password is in private key format
+        oldPrivateKey = deipRpc.auth.toWif(initiator.username, formPass.oldPassword, 'owner');
+        const oldPublicKey = deipRpc.auth.wifToPublic(oldPrivateKey);
+
+        // return if the public key from the password is not equal to the public key of the account
+        if (initiator.signUpPubKey !== oldPublicKey) throw new Error('Old password is invalid');
+      }
+    } catch (err) {
+      return Promise.reject(new Error(err));
+    }
```

Comment:
You don't need to reject promise inside catch. Also there is no need in this try-catch because error will be caught in ```.catch((err) => {
            this.emitError(err);
          })``` in updatePromise()